lib/util/state.js

```diff
@@ -25,32 +25,136 @@ export function getTimestamp (time = moment()) {
 }
 
 /**
- * Gets the OTP responses that contain errors. For batch routing this could be
- * an array of multiple responses, but for standard itinerary routing it will be
- * at most an array with one response.
+ * Analyzes all rental networks provided in the response and returns a list of
+ * those that have a FEED_WIDE error.
+ */
+export function getFeedWideRentalErrors (response) {
+  // return empty array if the plan key is not available in the response
+  if (!response) return []
+
+  // get first available rental info (this assumes that there won't be
+  // 2 rental info types (for example car rental access and vehicle
+  // rental egress))
+  const rentalInfo = response.bikeRentalInfo ||
+    response.carRentalInfo ||
+    response.vehicleRentalInfo
+  if (!rentalInfo) return []
+
+  // iterate through all networks and collect a list of feeds that have a feed-
+  // wide error
+  return Object.entries(rentalInfo)
+    .filter(([network, networkRentalInfo]) =>
+      networkRentalInfo.errors &&
+        networkRentalInfo.errors.some(error => error.severity === 'FEED_WIDE')
+    )
+    .map(([network, networkRentalInfo]) => ({
+      network,
+      networkRentalInfo
+    }))
+}
+
+/**
+ * Converts a comma-separated list of OTP mode values into a human-readable
+ * string
+ */
+function humanReadableMode (modeStr) {
+  if (!modeStr) return 'N/A'
+  const arr = modeStr.toLowerCase().replace(/_/g, ' ').split(',')
+  if (arr.length > 2) {
+    const last = arr.pop()
+    return arr.join(', ') + ' and ' + last
+  } else {
+    return arr.join(' and ')
+  }
+}
+
+/**
+ * Generates a list of issues with a list of OTP responses. For batch routing
+ * this could be an array of multiple responses, but for standard itinerary
+ * routing it will be at most an array with one response.
```

Comment:
Is this jsdoc correct? Wouldn't the array length be greater than one for standard itin routing if there was an OTP response error AND a feed-wide rental error?

Reply:
I updated the JSDoc accordingly in 8528d89154ec5e471d8ac40fc66ce4e35d974261